kubernetes.go

```diff
@@ -106,23 +118,32 @@ func (k *KubernetesRegistry) LookupService(ctx context.Context, name string) (Se
 	for _, obj := range k.store.List() {
 		pod, err := toPod(obj)
 		if err != nil {
-			events.Log("failed to covert data to pod: %{error}s", err)
+			events.Log("failed to convert data to pod: %{error}s", err)
 			continue
 		}
-
-		for _, container := range pod.Spec.Containers {
-			tags := []string{pod.Name}
-
-			for _, port := range container.Ports {
-				hosts = append(hosts, Host{
-					Addr: &net.TCPAddr{
-						IP:   net.ParseIP(pod.Status.PodIP),
-						Port: int(port.ContainerPort),
-					},
-					Tags: append(tags, port.Name),
-				})
+    // filtering pods based on podname, even if they are diff namepsaces for now, since the route for namespaces isnt made yet
+		if pod.Name == name {
```

Comment:
There is probably a way to do this directly in the Store API, no ? So we just get the POD(s) we want from the name by just querying the store.

Reply:
so the order is k.store -> informer.getstore -> listwatch . I looked into listwatch yesterday and decided against modifying what's stored in cache. put in a basic way of getting pods. Will look into informer and store further. informer has a GetByKey() function that may be useful.